### DISCUSSION TITLE
Clipping/cut text outside group

### DISCUSSION THREAD

**Discussion by mgierasinski96** [fabricjs/fabric.js#8956] [General]
I have drawn fabric.Group which consists of multiple rectangles.
```
const rectangles = []; 
for(let i = 0; i< 5; i++) {
rectangles.push(this.helper.drawRectangle(topPosition, leftPosition);
}
let group = new fabric.Group(rectangles, {
left:50,
top:50
});
```
next im adding text to this group
```
let offset = 40;
    var text = new fabric.Text('This is a very long text that is clipped', {
      fontSize: 40,
      top: -(group .height / 2 + offset),
      originX: 'center',
      originY: 'center'
    });
    
    group.add(text);
```
    
This is just an example code. I have additionaly some rotations but they are not important for a now.
What I get is a group with boundaries only around those rectangles. Text is positioned outside the group.
![image](https://github.com/fabricjs/fabric.js/assets/44431354/8b836b12-dd38-438f-9a9a-2ba632d141a9)
Whenever I'm moving or rotating a group text rotates as well but the origin of the group does not take into consideration text dimensions and this is and expected behaviour.
That's the main reason why I can't do 
```
// SOLUTION 1
let group = new fabric.Group([...rectangles, text] {
left:50,
top:50
});
```

In that case boundaries are around text as well and originX/Y equal to 'center' is not a center of rectangle group.

Now the problem is I also implemented logic for zooming. When I'm increasing zooming something weird happens with my text.
This is just being cut (horizontally and vertically)

![image](https://github.com/fabricjs/fabric.js/assets/44431354/18069c84-aea8-4cb8-8fb2-2fac9844ddea)

When i'm trying "SOLUTION 1" everything works fine so i think this is a problem with boundaries of group.
I have tried also changing clipPath for text and group but I believe this is not working.
Also tried addWithUpdate but it is completely destroying rest of my logic so i cannot go that way.
// SOLUTION 2
 group.add(text);
 group.addWithUpdate();
```
Do you have any other ideas how to prevent text from being cut?

**Comment by ShaMan123** [comment]
What version of fabric are you using?
Add a reproduction

**Reply by ShaMan123** [reply]
If not v6 that is most of your problem
